lib/rules/attributes-order.js

```diff
@@ -0,0 +1,231 @@
+          first_index: Math.min(...indexes),
+          unalphabetized_item_pos: alpha.findIndex((e, idx) => idx > 0 && e < alpha[idx - 1]),
+        };
+      }
+      return tokens;
+    }
+
+    function findUnalphabetizedElement(attributeType, nodeTokens) {
+      return nodeTokens[attributeType].items[nodeTokens[attributeType].unalphabetized_item_pos];
+    }
+
+    function findCurrentElement(attributeType, nodeTokens) {
+      return nodeTokens[attributeType].items.find(
+        ([index]) => index === nodeTokens[attributeType].last_index
+      );
+    }
+
+    function capitalizedAttribute(string) {
+      return string[0].toUpperCase() + string.slice(1, -1);
+    }
+
+    function createNotAlphabetizedErrorMessage(attributeType, source) {
+      return `${capitalizedAttribute(attributeType)} ${source} is not alphabetized`;
+    }
+
+    function createUnorderedErrorMessage(attributeType, source) {
+      const attributeOrder = config.attributeOrder.indexOf(attributeType);
+      if (attributeOrder === 0) {
+        const otherAttributes = `${config.attributeOrder[1]}, ${config.attributeOrder[2]} and ${config.attributeOrder[3]}`;
+
+        return `${capitalizedAttribute(attributeType)} ${source} must go before ${otherAttributes}`;
+      }
+      return `${capitalizedAttribute(attributeType)} ${source} must go after ${
+        config.attributeOrder[attributeOrder - 1]
+      }`;
+    }
+
+    function isAttributeUnordered(attributeType, nodeTokens) {
+      if (!nodeTokens[attributeType].exists) {
+        return false;
+      }
+      const attributeOrder = config.attributeOrder.indexOf(attributeType);
+      const hasAttributesAfterSplattributes =
+        nodeTokens[AttributeType.SPLATTRIBUTES].exists &&
+        nodeTokens[AttributeType.SPLATTRIBUTES].last_index <
+          nodeTokens[AttributeType.ATTRIBUTES].last_index;
+
+      switch (attributeOrder) {
+        case 0:
+          return (
+            nodeTokens[attributeType].last_index >
+              nodeTokens[config.attributeOrder[1]].first_index ||
+            nodeTokens[attributeType].last_index >
+              nodeTokens[config.attributeOrder[2]].first_index ||
+            nodeTokens[attributeType].last_index > nodeTokens[config.attributeOrder[3]].first_index
+          );
+
+        case 2:
+          if (attributeType !== AttributeType.MODIFIERS || !hasAttributesAfterSplattributes) {
+            return (
+              nodeTokens[attributeType].last_index < nodeTokens[config.attributeOrder[1]].last_index
+            );
+          }
+          break;
+
+        case 3:
+          if (attributeType !== AttributeType.MODIFIERS || !hasAttributesAfterSplattributes) {
+            return (
+              nodeTokens[attributeType].last_index < nodeTokens[config.attributeOrder[2]].last_index
+            );
+          }
+          break;
+
+        default:
+          break;
+      }
+    }
+
+    function checkMustacheAttributesOrder(node) {
```

Comment:
Can you add some high-level commenting to explain the general strategy for checking the order?